src/Analyzers/CSharp/Tests/Copilot/CSharpImplementNotImplementedExceptionDiagnosticAnalyzerTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Testing;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Copilot.UnitTests
+{
+    using VerifyCS = CSharpCodeFixVerifier<
+        CSharpImplementNotImplementedExceptionDiagnosticAnalyzer,
+        EmptyCodeFixProvider>;
+
+    public class CSharpImplementNotImplementedExceptionDiagnosticAnalyzerTests
+    {
+        [Fact]
+        public async Task TestThrowNotImplementedException()
+        {
+            var testCode = """
+                using System;
+
+                class C
+                {
+                    void M()
+                    {
+                        [|throw new NotImplementedException();|]
+                    }
+                }
+                """;
+
+            await new VerifyCS.Test
+            {
+                TestCode = testCode,
+                ExpectedDiagnostics =
+                {
+                    VerifyCS.Diagnostic().WithSpan(7, 9, 7, 45)
```

Comment:
shouldn't need this.  can use `{|DiagId:...|}` in the code to wrap where there's a diagnostic.